### PR TITLE
[Fleet] Add read privileges to profiling-* for symbolization support

### PR DESCRIPTION
**Pull request by vikmik**
This is a fix/followup to #95241 - we need read permissions on profiling indices for symbolization.
Related Kibana PR: [#155642](https://github.com/elastic/kibana/pull/155642/files)

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java [hunk 1]
```diff
@@ -69,11 +69,14 @@ final class ElasticServiceAccounts {
...
                         ".logs-endpoint.diagnostic.collection-*",
-                        ".logs-endpoint.action.responses-*",
-                        "profiling-*"
+                        ".logs-endpoint.action.responses-*"
                     )
                     .privileges("write", "create_index", "auto_configure")
                     .build(),
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices("profiling-*")
+                    .privileges("read", "write", "create_index", "auto_configure")
```
Comment (inge4pres) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java:78]: We don't need `create_index`, as the indices are managed by Fleet/Kibana

**Comment by danielmitterdorfer** [comment]
`elasticsearch-ci/cloud-deploy` was failing with

```
Could not find beats:metricbeat:8.9.0-SNAPSHOT.
  Searched in the following locations:
    - https://repo.maven.apache.org/maven2/beats/metricbeat/8.9.0-SNAPSHOT/maven-metadata.xml
    - https://repo.maven.apache.org/maven2/beats/metricbeat/8.9.0-SNAPSHOT/metricbeat-8.9.0-SNAPSHOT.pom
    - https://artifacts-snapshot.elastic.co/beats/8.9.0-SNAPSHOT/downloads/beats/metricbeat/metricbeat-8.9.0-SNAPSHOT-linux-x86_64.tar.gz
  Required by:
      project :distribution:docker
```

This is likely because there is no Beats artifact available yet after the version bump. So the error will persist until this is resolved.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java (hunk 1 above)
Comment (vikmik) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java]: I dropped it.

**Review by danielmitterdorfer — APPROVED** [review]
LGTM! Thanks for addressing this.

**Review by inge4pres — APPROVED** [review]
Thanks 🙏🏼

**Review by slobodanadamovic — APPROVED** [review]
LGTM

**Comment by inge4pres** [comment]
@elasticsearchmachine run elasticsearch-ci/cloud-deploy

**Pull request merged**